daemon/api_sideload_n_try_test.go

```diff
@@ -508,6 +509,58 @@ func (s *sideloadSuite) TestInstallPathSystemRestartImmediate(c *check.C) {
 	c.Check(systemRestartImmediate, check.Equals, true)
 }
 
+func (s *sideloadSuite) TestFormdataIsWrittenToCorrectTmpLocation(c *check.C) {
+	oldTempDir := os.Getenv("TMPDIR")
+	defer func() {
+		c.Assert(os.Setenv("TMPDIR", oldTempDir), check.IsNil)
+	}()
+	tmpDir := c.MkDir()
+	c.Assert(os.Setenv("TMPDIR", tmpDir), check.IsNil)
+
+	head := map[string]string{"Content-Type": "multipart/thing; boundary=--hello--"}
+	chgSummary, _ := s.sideloadCheck(c, sideLoadBodyWithoutDevMode, head, "local", snapstate.Flags{RemoveSnapPath: true})
+	c.Check(chgSummary, check.Equals, `Install "local" snap from file "a/b/local.snap"`)
+
+	files, err := ioutil.ReadDir(tmpDir)
+	c.Assert(err, check.IsNil)
+	c.Assert(files, check.HasLen, 0)
+
+	matches, err := filepath.Glob(filepath.Join(dirs.SnapBlobDir, dirs.LocalInstallBlobTempPrefix+"*"))
+	c.Assert(err, check.IsNil)
+	c.Assert(matches, check.HasLen, 1)
+
+	data, err := ioutil.ReadFile(matches[0])
+	c.Assert(err, check.IsNil)
+	c.Assert(string(data), check.Equals, "xyzzy")
+}
+
+func (s *sideloadSuite) TestSideloadMemoryLimit(c *check.C) {
+	s.daemonWithOverlordMockAndStore()
+
+	// check that there's a memory limit for the sum of the parts, not just each
+	bufs := make([][]byte, 2)
+	var body string
+
+	for i := range bufs {
+		bufs[i] = make([]byte, daemon.MaxReadBuflen/2+1)
+		_, err := rand.Read(bufs[i])
+		c.Assert(err, check.IsNil)
+
+		body += "--foo\r\n" +
+			"Content-Disposition: form-data; name=\"stuff\"\r\n" +
+			"\r\n" +
+			string(bufs[i]) +
+			"xyzzy\r\n"
```

Comment:
Is this intentional for this test to have a body with repeated content-disposition with same name?

Reply:
Yes, it shouldn't make a difference I think